api/app/models/land_change/tree_cover_loss.py

```diff
@@ -0,0 +1,114 @@
+    aoi: Annotated[AoiUnion, Field(discriminator="type")] = Field(
+        ...,
+        title="AOI",
+        description="AOI to calculate in.",
+    )
+    start_year: str = Field(
+        ...,
+        title="Start Date",
+        description="Must be year in YYYY date format. Minimum year is 2001.",
+        pattern=DATE_REGEX,
+        examples=["2020", "2024"],
+    )
+    end_year: str = Field(
+        ...,
+        title="End Date",
+        description="Must be year in YYYY date format. Minimum year is 2001.",
+        pattern=DATE_REGEX,
+        examples=["2023", "2024"],
+    )
+    canopy_cover: AllowedCanopyCover = Field(
+        ...,
+        title="Canopy Cover",
+    )
+    intersections: AllowedIntersections = Field(
+        ..., min_length=0, max_length=1, description="List of intersection types"
+    )
+
+    def thumbprint(self) -> uuid:
+        """
+        Generate a deterministic UUID thumbprint based on the model's JSON representation.
+
+        Returns:
+            uuid: UUID5 string derived from sorted JSON representation
+        """
+        # Convert model to dictionary with default settings
+        payload_dict = self.model_dump(
+            include=["aois", "start_year", "end_year", "canopy_cover", "intersections"]
+        )
+
+        payload_json = json.dumps(payload_dict, sort_keys=True)
+        return uuid.uuid5(uuid.NAMESPACE_DNS, payload_json)
+
+    @field_validator("start_year", "end_year")
+    def year_must_be_at_least_2001(cls, v: str) -> str:
+        year_int = int(v)
+        if year_int < 2001:
+            raise ValueError("Year must be at least 2001")
+        return v
+
+    @model_validator(mode="after")
+    def validate_year_range(self) -> "TreeCoverLossAnalyticsIn":
+        start = int(self.start_year)
+        end = int(self.end_year)
+        if end < start:
+            raise ValueError("end_year must be greater than or equal to start_year")
+        return self
+
+
+class TreeCoverLossAnalytics(StrictBaseModel):
+    result: Optional[dict] = {  # column oriented for loading into a dataframe
+        "__dtypes__": {
+            "country": "str",
+            "region": "int",
+            "subregion": "int",
+            "tree_cover_loss__year": "int",
+            "tree_cover_loss__ha": "float",
+            "gross_emissions_co2e_all_gases__mg": "float",
+        },
+        "country": ["BRA", "BRA", "BRA"],
+        "region": [1, 1, 1],
+        "subregion": [12, 12, 12],
+        "tree_cover_loss__year": [2022, 2023, 2024],
+        "tree_cover_loss__ha": [4045.406160862687, 4050.4061608627, 4045.406160862687],
+        "gross_emissions_co2e_all_gases__mg": [
+            3490821.6510292348,
+            114344.24741739516,
+            114347.2474174,
+        ],
+    }
+    metadata: Optional[dict] = None
+    message: Optional[str] = None
+    status: AnalysisStatus
+
+    model_config = {
+        "from_attributes": True,
+        "validate_by_name": True,
+    }
```

Comment:
Having a default result complete with actual data seems an interesting choice to me. Is there a particular situation this serves?

Reply:
Excellent catch! I made a proper example in commit: 8a7a82d5a03d3a686e5ce2294e9f0ab538e80286